pkg/plugins/secrets/k8s/store.go

```diff
@@ -91,9 +92,14 @@ func (s *KubernetesStore) Update(ctx context.Context, r core_model.Resource, fs
 		return errors.Wrap(err, "failed to convert core Secret into k8s counterpart")
 	}
 	secret.Namespace = s.namespace
-	if r.Descriptor().Scope == core_model.ScopeMesh {
-		setMesh(secret, r.GetMeta().GetMesh(), opts.Labels)
+
+	updateLabels := r.GetMeta().GetLabels()
+	if opts.ModifyLabels {
+		updateLabels = opts.Labels
 	}
+
+	setLabelsAnnotationsAndMesh(secret, r.GetMeta().GetMesh(), updateLabels)
```

Comment:
I guess the same issue as https://github.com/kumahq/kuma/pull/11940#discussion_r1823511171?

Reply:
Solved now